### DISCUSSION TITLE
Monsters ignore handheld sonar beacons inside subs

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#14453] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

A handheld sonar beacon turned on inside a submarine pings the area almost as well as a submarine's sonar turned on.
This makes the use of a ship's sonar almost unnecessary.
It seems to me that a sonar beacon turned on inside a boat should not sound outside the boat at all.
Just as a handheld sonar (turned on inside the ship) shows nothing outside the boat.

### Reproduction steps

Start game session
Take a handheld sonar beacon and turn it on inside sub.
Open navigation terminal.
Navigate with shop's sonar turned off
Monsters do not react to a submarine outside of direct line of sight.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
They both highlight things outside the ship, handheld sonar just have smaller radius

Ye, it's imba: you can ping molochs and endworms point blank without being spotted and see through sonar disruption
Also this guy https://github.com/FakeFishGames/Barotrauma/discussions/14386, still not fixed, heh

**Comment by gerakis** [comment]
I tried this and it didnt seem any better than running ship sonar. If anything the detection radius from hostile creatures was even larger. Also it makes sense you could see as its a form of noise, and any noise can be picked up by passive sonar.

**Reply by SomeRandomNoobKekeke** [reply]
They ignore it, are you sure you turned off your reactor?

https://github.com/user-attachments/assets/7970fc4a-a02f-4359-afbf-75fdd8d77961

Actually i think this might be not intended: handheldsonar has both sonar and provocative tags
![Screenshot_5](https://github.com/user-attachments/assets/39313e6a-863a-4d84-a6b3-ee892fb4e2e5)

But molochs shouldn't ignore sonars that are not in same sub
![Screenshot_6](https://github.com/user-attachments/assets/8b14b605-eef0-444e-8ef1-9e40f9c4169a)

Perhaps they should attack if any condition met, not ingnore if any

**Reply by Regalis11** [reply]
> Perhaps they should attack if any condition met, not ingnore if any

These are not conditions nor do they exclude each other.

**Comment by pavel4win** [comment]
Please note:
A handheld sonar beacon is a small battery-powered device.
In the hands of a diver outside the boat, it has a radius of about 50 meters of its built-in screen.
How can a handheld sonar beacon turned on inside a submarine illuminate an area 70 meters around?
Do the walls of the boat not muffle the sounds of a handheld sonar beacon at all?
If you put a handheld sonar beacon in your belt or in a closet, the illumination will continue.
Are the sounds of a handheld sonar beacon really so powerful that they pass through all the walls, closets and pockets?
Then why doesn't it hurt your ears?
Try an experiment:
Put the handheld sonar beacon in a closet and move 10 meters away from it. Do you stop hearing the handheld sonar beacon?
Then how can a submarine's sonar hear a sound of handheld sonar beacon that has traveled 70 meters underwater, reflected off a wall and returned 70 meters back?
It would be logical if a handheld sonar beacon illuminated the area only when it is outside the boat and also in your hand. If you hide it on your belt, it should be partially muffled.

Another oddity. The handheld sonar beacon is indestructible. The Doomworm can't swallow it. The Crawler can't bite it. After throwing the sonar beacon out of sub, I shoot the monsters of the abyss like in a shooting gallery, they aim at the beacon but can't do anything. I see the monsters in the light of the beacon and shoot at them, but they don't see me and circle around the beacon. Was that how it was intended?

This question is a little off topic, but I am interested:
If you run fast and shout loudly inside the submarine, will these sounds reach the walls and illuminate the area on the ship's sonar?

**Comment by Jasontti** [comment]
Tested the fix for this.
Sonar now attracts monsters properly and is not usable as navigation aid.
Closing discussion as resolved.

**Comment by pavel4win** [comment]
Do you wrote about fix "handheld sonar" or about "handheld sonar beacon" or both ?
